### PR TITLE
Reorganize repo and update build workflow

### PR DESCRIPTION
**Pull request by dplumb94**
The sawtooth-seth repo was separated from master by deleting out all non-Seth related documents and files. This has led to a disorganized and confusing directory structure. In order to give Seth a good user story and development workflow, this PR reorganizes the repo and simplifies the build process for development.

In order to build and run Seth, just run`docker-compose up --build`. This will build images and start up containers for all Seth components:
- Seth CLI
- Seth TP
- Block info TP
- Settings TP
- Sawtooth REST API
- Sawtooth Validator
- Seth JSON RPC

In order to accomplish this, the following steps were taken:
- Create new Dockerfiles for each Seth component
    - These Dockerfiles will import Sawtooth 1.0 dependencies instead of cloning in
      the whole repo
- Update docker-compose.yaml to build an image for each component and
  run the containers
- Update imports in .go files and Cargo.toml to correct dependencies

_Note: The Sawtooth Go SDK being used is an unofficial repo hosted at https://github.com/rberg2/sawtooth-go-sdk. Once there is an official hyperledger/sawtooth-sdk-go repo, we should switch to that._

### CODE REVIEW

File: bin/seth-protogen [hunk 1]
```diff
@@ -188,6 +158,24 @@ def fix_import(contents, pkg, sub_dir=False):
     )
 
 
+def protoc(src_dir, base_dir, pkg, language="python"):
+    if language == "python":
```
Comment (peterschwarz) [inline bin/seth-protogen]: Is there still any python in this repo?

Comment (dplumb94) [reply bin/seth-protogen]: Yes, some of the RPC tests use python and require the Seth protos

File: cli/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    curl \
+    git \
+    golang-1.9-go \
+    libssl-dev \
+    libzmq3-dev \
+    openssl \
+    python3 \
+    python3-grpcio-tools=1.1.3-1 \
+    python3-sawtooth-cli \
+ && curl -s -S -o /tmp/setup-node.sh https://deb.nodesource.com/setup_6.x \
```
Comment (peterschwarz) [inline cli/Dockerfile]: Not sure we need nodejs here

Comment (peterschwarz) [reply cli/Dockerfile]: May be removed in a future PR

Comment (peterschwarz) [reply cli/Dockerfile]: Nevermind, it looks used

**Pull request merged**